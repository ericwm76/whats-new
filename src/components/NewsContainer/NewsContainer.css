.main {
  display: grid;
  grid-area: main;
  grid-gap: 10px;
  grid-template-columns: 32% 32% 32%;
  grid-template-rows: auto auto auto;
  justify-content: center;
  /* margin-top: 50px; */
}