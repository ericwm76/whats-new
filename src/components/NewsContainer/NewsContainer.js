import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle'

const NewsContainer = ({local}) => {
  return (
    local.map(article => {
      return (
        <NewsArticle
          headline={article.headline} 
          img={article.img} 
          description={article.description}
          url={article.url}
          id={article.id}
          key={article.id}
        />
      )
    })
  )
}

export default NewsContainer;