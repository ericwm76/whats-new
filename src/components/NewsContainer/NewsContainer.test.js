import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer'

describe('NewsContainer', () =>{
  let mockArticles = [{ headline: "James joyce", img: "portrait of the artist", description: "as a young man"}, { headline: "Harry Potter", description: "Wizarding is cool", img: "hogwarts"}]
  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(<NewsContainer
      current={mockArticles}
    />)
    expect(wrapper).toMatchSnapshot();
  })
})