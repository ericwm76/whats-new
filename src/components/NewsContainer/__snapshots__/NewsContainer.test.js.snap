// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot 1`] = `
<div
  className="main"
>
  <NewsArticle
    description="as a young man"
    headline="James joyce"
    img="portrait of the artist"
    key="0"
  />
  <NewsArticle
    description="Wizarding is cool"
    headline="Harry Potter"
    img="hogwarts"
    key="1"
  />
</div>
`;
